templates/experiments/components/experiment_chat.html

```diff
@@ -189,39 +196,39 @@ <h3 class="font-bold text-lg">Translate Remaining Messages?</h3>
                     <div class="grow">
                       <div class="message-contents">
                         {% if message.translation == default_message %}
-                          <div id="chat-message-untranslated">
-                            <div class="htmx-show">
+                          <div id="chat-message-{{ message.id }}-translation">
+                            <div class="translation-loading-spinner-{{ message.id }}">
                               <span class="loading loading-spinner loading-sm"></span>
                               <span class="ml-2">Translating...</span>
                             </div>
-                            <div class="htmx-hide">
+                            <div class="translation-content-{{ message.id }}">
                               <p>{{ message.translation|render_markdown }}</p>
                             </div>
                           </div>
                         {% else %}
-                          <div id="chat-message-untranslated">
-                            <div class="htmx-hide">
+                          <div id="chat-message-{{ message.id }}-translation">
+                            <div class="translation-content-{{ message.id }}">
                               {% if message.translation %}
                                 <p>{{ message.translation|render_markdown }}</p>
                               {% else %}
                                 <p>{{ message.content|render_markdown }}</p>
                               {% endif %}
                             </div>
-                            <div class="htmx-show">
+                            <div class="translation-loading-spinner-{{ message.id }}" style="display: none;">
```

Comment:
nit: use `hidden` class instead of inline style

```suggestion
                            <div class="translation-loading-spinner-{{ message.id }} hidden">
```